cli_tools/gce_vm_image_import/importer/importer_test.go

```diff
@@ -352,6 +352,72 @@ func TestRun_ProcessCantTimeoutImportSucceeds(t *testing.T) {
 	assert.True(t, duration > importer.timeout)
 }
 
+func TestRunStep_VeryShortTimeout(t *testing.T) {
+	// This test ensures that inflater.runStep doesn't dead lock when timeout
+	// has already passed before step function is able to run.
+
+	doTestWithTimeOut(t, 2*time.Second, func(t *testing.T) {
+		// run this test with a timeout as it might never finish if there is a bug
+
+		importer := importer{
+			preValidator:      mockValidator{},
+			inflater:          &mockInflater{},
+			processorProvider: &mockProcessorProvider{},
+			loggableBuilder:   service.NewSingleImageImportLoggableBuilder(),
+			// this simulates a very short timeout, or a situation when a timeout
+			// occurs immediately after one step finishes and the next one is about
+			// to start
+			timeout: 0 * time.Millisecond,
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), importer.timeout)
+		defer cancel()
+
+		cancelChan := make(chan bool)
+		importer.runStep(ctx,
+			func() error {
+				// step
+				select {
+				case <-cancelChan:
+					break
+				case <-time.After(time.Second * 5):
+					break
+				}
+				return nil
```

Comment:
Is it worth asserting that the step isn't run? (And that the cancellation is run?) Or does that make the test too fragile?

Reply:
The main purpose of the test is to make sure there are no deadlocks on timeout. However, in this case (0ms timeout) it makes sense asserting step didn't actually run, so I added the check.